src/pipx/venv.py

```diff
@@ -327,8 +342,40 @@ def list_installed_packages(self) -> Set[str]:
         pip_list = json.loads(cmd_run.stdout.strip())
         return set([x["name"] for x in pip_list])
 
+    def _find_entry_point(self, app: str) -> Optional[EntryPoint]:
+        if not self.python_path.exists():
+            return None
+        site_packages = get_site_packages(self.python_path)
+        for dist in Distribution.discover(path=[str(site_packages)]):
```

Comment:
How about limiting our check for the `pipx.run` entrypoint to only the main package?

Otherwise if we found a dependency with a `pipx.run` entrypoint having the app name, we would match it instead.  This sounds like a dumb thing to happen, but I've seen dumb things in the past (like a package using two dependencies that both use the same name for console scripts).

Reply:
Good idea, added.

Reply:
Hmm, this does not really work well after I think about this more, because `pipx run` does not really know what the “main package” is in a cached venv. It’s wrong to use the `app` argument, that would make usages like `pipx run tox4 --spec "tox>=4a0"` fail.

Reply:
Don't we have that information in the metadata?  e.g. `self.main_package_name`.

Reply:
You could even use `pipx.venv_inspect.get_dist` using `self.main_package_name` as the first argument.

Reply:
Oh, I didn’t know that! Let me try it out.

Reply:
That seems to work. PR updated.

Reply:
Is `importlib.metadata.Distribution.discover()` documented anywhere?  I'm only finding it by browsing the `importlib.metadata` source.  It would probably be useful to use in `pipx.venv_inspect`.

Reply:
Also, just out of curiosity, is there a way you would get multiple `dists` from from `Distribution.discover()` if you searched for the name of the package?

Reply:
I don’t think `Distribution.discover()` is documented. The author told me about it when I was investigating the possibility of using importlib.metadata in pip (to replace `pkg_resources`).

> Is there a way you would get multiple dists from from Distribution.discover() if you searched for the name of the package?

Yes, because it is possible to have multiple installations of a package in different `sys.path` entries. For example:

```
pip install build
pip install --user build
```

would give you two `build` distributions, one under the system site-packages, and the other in the [per-user site-packages](https://www.python.org/dev/peps/pep-0370/). Another possible case is [enabling system site-packages](https://www.python.org/dev/peps/pep-0405/#isolation-from-system-site-packages) in a virtual environment